app/src/main/java/com/android/unio/ui/home/Home.kt

```diff
@@ -102,152 +90,199 @@ fun HomeScreen(
-                    },
-                    onSearch = {},
-                    expanded = false,
-                    onExpandedChange = {},
-                    placeholder = {
-                      Text(text = context.getString(R.string.explore_search_placeholder))
-                    },
-                    trailingIcon = {
-                      if (searchState == SearchViewModel.Status.LOADING) {
-                        CircularProgressIndicator()
-                      } else {
-                        Icon(
-                            Icons.Default.Search,
-                            contentDescription =
-                                context.getString(R.string.home_content_description_search_icon))
-                      }
-                    },
-                )
-              },
-              expanded = false,
-              onExpandedChange = {},
-              modifier = Modifier.padding(horizontal = 16.dp).testTag(HomeTestTags.SEARCH_BAR)) {}
+        val list =
+            listOf(
+                context.getString(R.string.home_tab_all),
+                context.getString(R.string.home_tab_following))
+        val scope = rememberCoroutineScope()
+        val pagerState = rememberPagerState(initialPage = 0) { nbOfTabs }
 
-          // Event List
-          if (searchQuery.isNotEmpty() &&
-              (searchState == SearchViewModel.Status.SUCCESS ||
-                  searchState == SearchViewModel.Status.LOADING)) {
-            if (searchResults.isEmpty()) {
+        val sizeList = remember { mutableStateMapOf<Int, Pair<Float, Float>>() }
+
+        val progressFromFirstPage by remember {
+          derivedStateOf { pagerState.currentPageOffsetFraction + pagerState.currentPage.dp.value }
+        }
+        val colorScheme = MaterialTheme.colorScheme
+
+        // Tab Menu at the top of the page
+        TabRow(
+            selectedTabIndex = pagerState.currentPage,
+            contentColor = Color(0xFF362C28),
+            divider = {},
+            indicator = {
+              // method that draws the indicator bar below the tab menu
               Box(
-                  modifier = Modifier.fillMaxSize().padding(paddingValues),
-                  contentAlignment = Alignment.Center) {
-                    Text(context.getString(R.string.explore_search_no_results))
-                  }
-            } else {
-              LazyColumn(
-                  contentPadding = PaddingValues(vertical = 8.dp),
-                  modifier = Modifier.fillMaxSize().padding(horizontal = 32.dp)) {
-                    items(searchResults) { event ->
-                      EventCard(navigationAction, event, userViewModel, eventViewModel)
+                  modifier =
+                      Modifier.fillMaxSize().drawBehind {
+                        val tabWidth = sizeList[0]!!.first
+                        val height = sizeList[0]!!.second
```

Comment:
Instead of asserting non null, you can add a default value. Because you cannot ensure that this list have non-null element